### PR TITLE
Integrate BanyanDB cluster as storage solution

### PR DESCRIPTION
**Pull request by ButterBright**

### CODE REVIEW

**Comment by wu-sheng** [comment]
Default values of `standalone.image.repository` and tags in https://github.com/apache/skywalking-banyandb-helm should be official releases before the integration.

File: chart/skywalking/Chart.yaml [hunk 1]
```diff
@@ -39,6 +39,6 @@ dependencies:
     repository: https://raw.githubusercontent.com/bitnami/charts/archive-full-index/bitnami
     condition: postgresql.enabled
   - name: skywalking-banyandb-helm
```
Comment (kezhenxu94) [inline chart/skywalking/Chart.yaml:41]: Let's rename this to skywalking-banyandb

**Review by kezhenxu94 — APPROVED** [review]
Overall looks good

**Review by kezhenxu94 — APPROVED** [review]
Thanks

File: chart/skywalking/Chart.yaml [hunk 2]
```diff
@@ -39,6 +39,7 @@ dependencies:
...
-    version: 0.1.0
-    repository: oci://registry-1.docker.io/apache
+    alias: banyandb
+    version: 0.0.0-8bdff75
+    repository: oci://ghcr.io/apache/skywalking-banyandb-helm
```
Comment (wu-sheng) [inline chart/skywalking/Chart.yaml:44]: You should point to docker hub, rather than ghcr. Why is this changed?

Comment (wu-sheng) [reply chart/skywalking/Chart.yaml:44]: Could you release 0.2.0 of the banyandb helm?

Comment (wu-sheng) [reply chart/skywalking/Chart.yaml:44]: Meanwhile, where is ` oci://registry-1.docker.io/apache`? Is this repository existing?

Comment (ButterBright) [reply chart/skywalking/Chart.yaml:44]: We have not yet released banyandb and banyandb-helm, so I temporarily pointed the repository to ghcr. I will update these values together after the releases are completed.

Comment (wu-sheng) [reply chart/skywalking/Chart.yaml:44]: We have the previous(0.1.0) version, https://github.com/apache/skywalking-banyandb-helm.

Are you going to release that soon? Otherwise, once skywalking-helm 4.6 is releasing, it will carry the unreleased BanyanDB version(some snapshot version). This is not correct in the ASF.

Comment (ButterBright) [reply chart/skywalking/Chart.yaml:44]: As far as I know, we will release shortly after the completion of testing on banyandb, and it should be relatively fast.

Comment (wu-sheng) [reply chart/skywalking/Chart.yaml:44]: Could you post an issue on Helm 4.6.0 milestone to track this, and you should fix [this](https://github.com/apache/skywalking-helm/pull/145#issuecomment-2021862061)

Comment (wu-sheng) [reply chart/skywalking/Chart.yaml:44]: @hanahmily told me, your deployment should not care about BanyanDB 0.5 or 0.6(dev or unreleased). So, this should not be a blocker for you. 
I know 0.6 is much better, so you only need to keep all in 0.5 by default, and upgrade to 0.6 once it is released.

Comment (ButterBright) [reply chart/skywalking/Chart.yaml:44]: Okay.

Comment (ButterBright) [reply chart/skywalking/Chart.yaml:44]: > @hanahmily told me, your deployment should not care about BanyanDB 0.5 or 0.6(dev or unreleased). So, this should not be a blocker for you. I know 0.6 is much better, so you only need to keep all in 0.5 by default, and upgrade to 0.6 once it is released.

There is a [feature](https://github.com/apache/skywalking-banyandb/pull/331) related to etcd authentication that has not been implemented in version 0.5. It seems that we still need to rely on the upcoming BanyanDB 0.6 version. After discussing with @hanahmily , we decide to change these tags later on.
